NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+ # First merge in ndvi scores
+ 
+ nigerShpMerged_admin2 =
+   nigerMapAdmin2 %>% 
+   full_join(admin2ndvi, by=c("admin2Pcod" = "admin2Pcod")) 
+ 
+ # also merge in admin3
+ nigerShpMerged_admin3 =
+   nigerMapAdmin3 %>% 
+   full_join(admin3ndvi, by=c("adm_03" = "adm_03"))
+ # Generate Z-Score Maps (Admin 2 and 3) -----
+ nigerShpMerged_admin2 %>% 
+   filter(year %in% c(2011, 2014, 2015, 2017, 2018)) %>% 
+   ggplot() + 
+   geom_sf(aes(fill = zscore_ndvi),color = NA, alpha = 0.8) +
+   scale_fill_viridis_c(direction = -1) +
+   facet_wrap(~year, nrow = 1) +
+   labs(title="Annual Z-Score NDVI by Department (Admin 2)", fill = "z-score" ) + 
+   theme_classic() + 
+   theme(axis.text.x = element_blank(),
+         axis.text.y = element_blank(),
+         axis.ticks = element_blank(),
+         rect = element_blank())
+ 
+  nigerShpMerged_admin3 %>% 
+    filter(year %in% c(2011, 2014, 2015, 2017, 2018)) %>% 
+             ggplot() + 
+             geom_sf(aes(fill = zscore_ndvi),color = NA, alpha = 0.8) +
+             scale_fill_viridis_c(direction = -1) +
+             facet_wrap(~year, nrow = 1) +
+             labs(title="Annual Z-Score NDVI by  Commune (Admin 3)", 
+                  fill = "z-score" ) + 
+             theme_classic() + 
+             theme(axis.text.x = element_blank(),
+                   axis.text.y = element_blank(),
+                   axis.ticks = element_blank(),
+                   rect = element_blank())
+
+
+ 
+ # All Years 3 Regions
+ 
+ Desert = (yearData_admin2$admin1Name == "Agadez"|yearData_admin2$admin1Name == "Diffa"|yearData_admin2$admin1Name == "Zinder")
+ agroPastoral = (yearData_admin2$admin1Name == "Tahoua"|yearData_admin2$admin1Name == "Maradi")
+ rainfed = (yearData_admin2$admin1Name == "Tillabery" | yearData_admin2$admin1Name == "Dosso"|| yearData_admin2$admin1Name == "Niamey")
+ 
+ # Filtered 1981 to 2010 Baseline Data ----
+ 
+  baselineData_admin2 <- yearData_admin2 %>% 
+    group_by(admin2Name, admin2Pcod) %>%
+    filter(year >= "1981" & year <= "2010") %>% 
+    summarise(baselineTotal_ndvi = sum(peak_ndvi, na.rm = TRUE),
+              baselineMean_ndvi = mean(peak_ndvi, na.rm = TRUE),
+              baselineSD_ndvi = sd(peak_ndvi, na.rm = TRUE)) %>%
+    ungroup()
+  
+  baselineData_admin3 <- yearData_admin3 %>% 
+    group_by(adm_03, rowcacode3) %>%
+    filter(year >= "1981" & year <= "2010") %>% 
+    summarise(baselineTotal_ndvi = sum(peak_ndvi, na.rm = TRUE),
+              baselineMean_ndvi = mean(peak_ndvi, na.rm = TRUE),
+              baselineSD_ndvi = sd(peak_ndvi, na.rm = TRUE)) %>%
+    ungroup()
+  
+  
+ # # 2011 to 2020 Filtered with baseline z-score
+  periodData_admin2 <- yearData_admin2 %>% 
+    group_by(admin2Name,admin2Pcod, year) %>%
+    filter(year >= "2011" & year <= "2021") %>% 
+    ungroup()
+ 
+  
+ 
+  datajoin_admin2 = merge(x = baselineData_admin2, y = periodData_admin2, by = "admin2Pcod") %>%
```

Comment:
merge is base R syntax -- preferable to use tidyverse syntax (leftjoin or fulljoin here) for consistency